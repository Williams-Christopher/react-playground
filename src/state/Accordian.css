.Accordian-list {
    list-style: none;
    font-family: sans-serif;
    border: 1px solid black;
    width: 75%;
    margin: 10px auto;
}

.Accordian-list-item {
    margin: 10px;
}

.Accordian-list-button {
    border: 1px solid black;
    min-width: 5%;
    padding: 5px;
}

.Accordian-list-item-content {
    background: whitesmoke;
    font-style: italic;
}