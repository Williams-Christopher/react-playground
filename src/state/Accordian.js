import React from 'react';
import './Accordian.css'

export default class Accordian extends React.Component {
    static defaultProps = {
        sections: [],
    }

    state = {
        displaySection: null,
    }

    generateSections() {
        return this.props.sections.map((section, index) => {
            return (
                <li key={index} className='Accordian-list-item'>
                    <button key={index} className='Accordian-list-button' onClick={() => this.handleClick(index)}>
                        {section.title}
                    </button>
                    {index === this.state.displaySection ? <p className='Accordian-list-item-content'>{section.content}</p> : ''}
                </li>
            )
        });
    }

    handleClick(index) {
        if (index === this.state.displaySection) {
            index = null;
        }

        this.setState({
            displaySection: index,
        })
    }

    render() {
        return(
            <ul className='Accordian-list'>
                {this.generateSections()}
            </ul>
        )
    }
}