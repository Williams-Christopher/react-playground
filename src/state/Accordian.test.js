import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

import Accordian from './Accordian';

describe('Accordian Component Tests', () => {
    const sections = [
        {
            title: 'Section 1',
            content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
            title: 'Section 2',
            content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
            title: 'Section 3',
            content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
    ]
      
    it('Renders without an error', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordian />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('Initially renders with all sections closed', () => {
        const wrapper = shallow(<Accordian sections={sections}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('Renders only the section whose button was clicked', () => {
        const wrapper = shallow(<Accordian sections={sections} />);
        wrapper.find('button').at(0).simulate('click');
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('Closes a previously open section when a different section button is clicked', () => {
        const wrapper = shallow(<Accordian sections={sections} />);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
