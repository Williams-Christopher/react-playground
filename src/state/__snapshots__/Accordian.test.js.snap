// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component Tests Closes a previously open section when a different section button is clicked 1`] = `
<ul
  className="Accordian-list"
>
  <li
    className="Accordian-list-item"
    key="0"
  >
    <button
      className="Accordian-list-button"
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="Accordian-list-item"
    key="1"
  >
    <button
      className="Accordian-list-button"
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p
      className="Accordian-list-item-content"
    >
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="Accordian-list-item"
    key="2"
  >
    <button
      className="Accordian-list-button"
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component Tests Initially renders with all sections closed 1`] = `
<ul
  className="Accordian-list"
>
  <li
    className="Accordian-list-item"
    key="0"
  >
    <button
      className="Accordian-list-button"
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="Accordian-list-item"
    key="1"
  >
    <button
      className="Accordian-list-button"
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    className="Accordian-list-item"
    key="2"
  >
    <button
      className="Accordian-list-button"
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component Tests Renders only the section whose button was clicked 1`] = `
<ul
  className="Accordian-list"
>
  <li
    className="Accordian-list-item"
    key="0"
  >
    <button
      className="Accordian-list-button"
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <p
      className="Accordian-list-item-content"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    className="Accordian-list-item"
    key="1"
  >
    <button
      className="Accordian-list-button"
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    className="Accordian-list-item"
    key="2"
  >
    <button
      className="Accordian-list-button"
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
